### PR TITLE
Adding support for additional interceptors

### PR DESCRIPTION
**Pull request by ankurs**

### CODE REVIEW

File: interceptors.go [hunk 1]
```diff
@@ -44,30 +50,64 @@ func FilterMethodsFunc(ctx context.Context, fullMethodName string) bool {
...
+
+// AddStreamServerInterceptor adds a server interceptor to default server interceptors
+func AddStreamServerInterceptor(ctx context.Context, i grpc.StreamServerInterceptor) {
+	streamServerInterceptors = append(streamServerInterceptors, i)
+}
+
+// ReplaceColdBrewServerInterceptors allows enabling/disabling coldbrew server interceptors
+//
+// when set to true, the coldbrew server interceptors will not used
+func ReplaceColdBrewServerInterceptors(ctx context.Context, flag bool) {
```
Comment (svetha-cvl) [inline interceptors.go]: Thanks for this change @ankurs
Suggestion: Can we consider making the new Add methods variadic and accept a list of interceptors? Clients can set DefaultInterceptors and/or their custom interceptors and we can avoid the replaceServerInterceptors flag. i.e. all desired interceptors, whether default or custom, are added explicitly by clients and coldbrew uses only the added interceptors.
Advantage: Behaviour becomes explicitly opt-in from clients and we do not need the replace logic.

However this is a !! BREAKING !! change for clients who upgrade coldbrew expecting the defaults to work out of the box. Not sure if that is acceptable.

File: interceptors.go [hunk 2]
```diff
@@ -44,30 +50,64 @@ func FilterMethodsFunc(ctx context.Context, fullMethodName string) bool {
...
+
+// AddStreamClientInterceptor adds a server interceptor to default server interceptors
+func AddStreamClientInterceptor(ctx context.Context, i grpc.StreamClientInterceptor) {
+	streamClientInterceptors = append(streamClientInterceptors, i)
+}
+
+// ReplaceColdBrewClientInterceptors allows enabling/disabling coldbrew client interceptors
+//
+// when set to true, the coldbrew client interceptors will not used
+func ReplaceColdBrewClientInterceptors(ctx context.Context, flag bool) {
```
Comment (svetha-cvl) [inline interceptors.go]: Can we rename this to UseColdBrewClientInterceptors or UseDefaultClientInterceptors instead for readability?

**Review by svetha-cvl — CHANGES_REQUESTED** [review]

File: interceptors.go (hunk 1 above)
Comment (ankurs) [reply interceptors.go]: I have made the functions variadic, typically we don't expect majority of services to need custom interceptors and we just provide a sane default for them to work with

on the default interceptors I thinks its better to make them opt-in, with the ability to disable them which also preserves the current behaviour

File: interceptors.go (hunk 2 above)
Comment (ankurs) [reply interceptors.go]: done

File: interceptors.go (hunk 1 above)
Comment (svetha-cvl) [reply interceptors.go]: Makes sense 👍

**Review by svetha-cvl — APPROVED** [review]

**Pull request merged**